src/Jackett.Common/Indexers/Corsarored.cs

```diff
@@ -1,256 +1,247 @@
+			var releases = await PerformQuery(new TorznabQuery());
+
+			await ConfigureIfOK(string.Empty, releases.Any(),
+				() => throw new Exception("Could not find release from this URL."));
+
+			return IndexerConfigurationStatus.Completed;
+		}
+
+		private dynamic CheckResponse(WebClientStringResult result)
+		{
+			try
+			{
+				var json = JsonConvert.DeserializeObject<dynamic>(result.Content);
+
+				switch (json)
+				{
+					case JObject _ when json["ok"] != null && (bool) json["ok"] == false:
+						throw new Exception("Server error");
+					default:
+						return json;
+				}
+			}
+			catch (Exception e)
+			{
+				logger.Error("checkResponse() Error: ", e.Message);
+				throw new ExceptionWithConfigData(result.Content, configData);
+			}
+		}
+
+		private async Task<dynamic> SendApiRequest(IEnumerable<KeyValuePair<string, string>> data)
+		{
+			//var jsonData = JsonConvert.SerializeObject(data);
+			var result = await PostDataWithCookiesAndRetry(ApiSearch, data, null, SiteLink, _apiHeaders, null, true);
+			return CheckResponse(result);
+		}
+
+		private async Task<dynamic> SendApiRequestLatest()
+		{
+			var result = await RequestStringWithCookiesAndRetry(ApiLatest, null, SiteLink, _apiHeaders);
+			return CheckResponse(result);
+		}
+
+		private string GetApiCategory(TorznabQuery query)
+		{
+			var cats = MapTorznabCapsToTrackers(query);
+			var apiList = new List<string>();
+			foreach (var apiCat in cats.Select(cat =>
+				_apiCategories.FirstOrDefault(kvp => kvp.Value.Contains(cat)).Key))
+			{
+				if (apiCat == "0") return apiCat;
+				if (!apiList.Contains(apiCat))
+					apiList.Add(apiCat);
+				if (apiList.Count > 1) return "0";
+			}
+
+			return apiList[0];
+		}
+
+		protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query)
+		{
+			var releases = new List<ReleaseInfo>();
+			var searchString = query.GetQueryString();
+
+			if (string.IsNullOrWhiteSpace(searchString))
+			{
+				// no term execute latest search
+				var result = await SendApiRequestLatest();
+
+				try
+				{
+					// this time is a jarray
+					var json = (JArray) result;
+
+					releases.AddRange(json.Select(MakeRelease));
+				}
+				catch (Exception ex)
+				{
+					OnParseError(result.ToString(), ex);
+				}
+
+				return releases;
+			}
+
+			var queryCollection = new Dictionary<string, string>
+			{
+				["term"] = searchString,
+				["category"] = GetApiCategory(query)
+			};
+
+			for (var page = 1; page <= MaxSearchPageLimit; page++)
+			{
+				// update page number
+				queryCollection["page"] = page.ToString();
+
+				var result = await SendApiRequest(queryCollection);
+				try
+				{
+					// this time is a jobject
+					var json = (JObject) result;
+
+					// check JSON response and exit if no results (!x.Any())
+					// throws exception if json["results"] is null or not a JArray
+					if (!(json["results"] as JArray ?? throw new Exception("Error invalid JSON response")).Any())
```

Comment:
instead of that, you can check if the number of results = 25 to avoid asking for an extra page.
i did that here => https://github.com/Jackett/Jackett/blob/61dc4ade17939c70a393e47ec9707597ef233d5b/src/Jackett.Common/Indexers/SolidTorrents.cs#L27

Reply:
There's a tradeoff in both. If the API changes how many results are returned, then it could break checking more than just page 1. Mine was just based off what was already there, and I can see it both ways. I'll go ahead and change it.